### ISSUE TITLE
Upscale Image (using Model) crashes before sending output

### ISSUE THREAD

**Issue by ricperry** [comfyanonymous/ComfyUI#3240]
My batch image upscale is crashing just prior to sending the output to the next node. ComfyUI crashes without outputting any error messages. My usage situation follows:

140 frames
576 x 576 res 16bit pngs
4x_NMKD-Siax_200k.pth upscaler
standalone workflow (no SD models loaded, just a plain pixel space upscaling workflow after a system reboot)
 
![image](https://github.com/comfyanonymous/ComfyUI/assets/34995372/8bd26e52-1aa3-4f6a-9222-850a06c8d0f8)

**Comment by WilliamStanford** [comment]
Getting the same problem, any luck solving it?

**Comment by ltdrdata** [comment]
Monitor to see if it's not a lack of memory.

**Comment by WilliamStanford** [comment]
> Monitor to see if it's not a lack of memory.

That's what I figured it would be, only happens when trying to upscale many frames at once. Any tips on how to get around this? 

I was thinking I could try split the images into batches, and then recombine them, but I'm not quite sure how to do that using ComfyUI.

**Comment by ricperry** [comment]
Yes, it exceeded my system ram + swap, then the process just quits. My VRAM was only at ~60%. I guess I'll have to expand the size of my swap file.

**Comment by ltdrdata** [comment]
In such cases, I recommend using a node that fetches images sequentially one by one from the folder, rather than fetching all at once, and processing them separately with an auto queue.
From what I remember, there is such a node in the WAS node.

**Comment by AgeOfAlgorithms** [comment]
I'm having the exact same issue but with video upscaling ksampler. Did anyone solve this problem?

**Comment by revolvedai** [comment]
Seems to be an issue across all these upscale nodes, I have tried them all. Tried increasing swap, RAM, and still having the same issue. Big memory spike at last minute kills the server.

**Comment by PeterTucker** [comment]
Bump. Anyone find a fix for this?

**Comment by ltdrdata** [comment]
> Bump. Anyone find a fix for this?

Please check the number of image batches and the resolution you are upscaling to. It is likely that you are trying to process data that is too large and too much for your system RAM to handle.

**Comment by ricperry** [comment]
The point of this node is useless if it can't dynamically determine how to manage system memory. It's not like I'm trying to upscale a 15sec 4k/60p video. Maybe only keep 4 frames resident in memory at a time with a sliding context window if it needs. Otherwise, why wouldn't it just upscale 1 image at a time and save the frame?

**Comment by ltdrdata** [comment]
> The point of this node is useless if it can't dynamically determine how to manage system memory. It's not like I'm trying to upscale a 15sec 4k/60p video. Maybe only keep 4 frames resident in memory at a time with a sliding context window if it needs. Otherwise, why wouldn't it just upscale 1 image at a time and save the frame?

That's why I suggested converting the image batch to a image list for processing above.
